packages/mongo-nav/src/mongo-select/OrgSelect.tsx

```diff
@@ -0,0 +1,225 @@
+`;
+
+const productStyle = css`
+  font-size: 12px;
+  color: ${uiColors.gray.dark2};
+  font-weight: bolder;
+  white-space: nowrap;
+`;
+
+const orgOptionContainerStyle = css`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const noOrgStyle = css`
+  font-size: 14px;
+  padding: 4px 8px;
+  margin-bottom: 20px;
+`;
+
+const allOrgLinkStyle = css`
+  color: ${uiColors.blue.base};
+  text-decoration: none;
+  &:visited {
+    color: ${uiColors.blue.base};
+    text-decoration: none;
+  }
+`;
+
+const formattedPlanTypes: Record<PlanType, string> = {
+  [PlanType.Atlas]: 'Atlas',
+  [PlanType.Cloud]: 'Cloud Manager',
+  [PlanType.OnPrem]: 'Ops Manager',
+} as const;
+
+interface OrganizationMongoSelectProps extends BaseMongoSelectProps {
+  data?: Array<OrganizationInterface>;
+  current?: CurrentOrganizationInterface;
+  constructOrganizationURL: NonNullable<
+    MongoNavInterface['constructOrganizationURL']
+  >;
+  isOnPrem?: boolean;
+  isActive?: boolean;
+  disabled?: boolean;
+}
+
+function OrgSelect({
+  current,
+  data = [],
+  urls = {},
+  isActive,
+  onChange: onChangeProp,
+  onClick: onClickProp,
+  constructOrganizationURL,
+  isOnPrem,
+  disabled,
+  loading,
+  className,
+}: OrganizationMongoSelectProps) {
+  const [value, setValue] = useState('');
+  const [open, setOpen] = useState(false);
+  const onElementClick = useOnElementClick();
+
+  useEffect(() => {
+    if (!open) {
+      setValue('');
+    }
+  }, [open]);
+
+  let renderedData = data;
+
+  const filterData = () => {
+    const sanitizedValue = value.replace(/\\/g, '\\\\');
+    const search = new RegExp(String(sanitizedValue), 'i');
+
+    const filtered = data?.filter(datum => {
+      return search.test(datum.orgName);
+    });
+
+    return filtered;
+  };
+
+  if (!onChangeProp) {
+    renderedData = filterData();
+  }
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = e.target.value.replace(/\\/g, '\\');
+
+    setValue(val);
+
+    if (onChangeProp) {
+      return onChangeProp(value, e);
+    }
+  };
+
+  const onClick = (e: React.MouseEvent) => {
+    setOpen(false);
+    return onClickProp?.(e);
+  };
+
+  const renderOrganizationOption = (datum: OrganizationInterface) => {
+    const { orgId, orgName, planType } = datum;
+    const isActive = orgId ? orgId === current?.orgId : false;
+
+    return (
+      <MenuItem
+        data-testid="org-option"
+        key={orgId}
+        active={isActive}
+        className={menuItemContainerStyle}
+        onClick={onClick}
+        href={constructOrganizationURL(datum)}
+      >
+        <div className={orgOptionContainerStyle}>
+          <span className={nameStyle}>
+            {orgName} {isActive && '(current)'}
+          </span>
+
+          {!isOnPrem && (
+            <span className={productStyle}>{formattedPlanTypes[planType]}</span>
+          )}
+        </div>
+      </MenuItem>
+    );
+  };
+
+  return (
+    <OrgTrigger
```

Comment:
This is a nit, and I'm happy either way but now that we consolidating should we just inline the Trigger functions directly into the component, rather than breaking them out as their own component? Either way the readability has gone up 1000% so very grateful for this PR.

Reply:
Great idea.